### PR TITLE
CI: print expected formatter output on failure

### PR DESCRIPTION
**Pull request by 678098**
https://github.com/bloomberg/blazingmq/actions/runs/8345560716/job/22840749145

The example formatting check failure with the desired output.

As a bonus, we remove an extra dependency and the check takes 6 seconds now instead of 6 minutes.

### CODE REVIEW

**Comment by 678098** [comment]
https://github.com/bloomberg/blazingmq/actions/runs/8345560716/job/22840749145

The example formatting check failure

File: .github/workflows/formatting-check.yaml
Comment (pniedzielski) [inline .github/workflows/formatting-check.yaml:22]: Do we want to diff against `origin/main` or against the common ancestor of `origin/main` and this branch ([`git merge-base`](https://git-scm.com/docs/git-merge-base))?  Not sure which is right in this context, especially when you take into account merge commits in a PR or `main` getting changes of its own.

**Review by pniedzielski — CHANGES_REQUESTED** [review]
I am so very on-board with this change, there was no need to use a pre-cooked Docker-container action for this.  One question in the `formatting-check.yaml` changes, but if the behavior as submitted is right, rerequest a review and I'll approve.

File: .github/workflows/formatting-check.yaml
Comment (678098) [reply .github/workflows/formatting-check.yaml:22]: For this, I believe we can keep things simple now.
What matters, if the code going to `origin/main` is formatted in the end. It might not work as intended for complex merges from non-main branch to other non-main branch, but it should be fine because at one point we'll decide to merge that non-main branch to origin/main and all formatter problems will arise to fix.

Personally, I prefer this fix to be able to do small patches more easily, without need to switch branches to run clang-format. It might also help with external contributions.

Comment (pniedzielski) [reply .github/workflows/formatting-check.yaml:22]: Let's keep it as is now then.  I'm still not sure what will happen if `main` gets ahead of the PR branch, but we'll have to do a rebase anyway before merging, so it should be fine.  I'll dismiss my review and make a new approve one.

**Review by pniedzielski — APPROVED** [review]

**Pull request merged**
